opp_gui/src/utils.cpp

```diff
@@ -31,24 +32,17 @@ namespace utils
 {
 namespace vm = visualization_msgs;
 
-bool getMeshMsgFromResource(const std::string& resource, shape_msgs::Mesh& mesh_msg)
+bool getMeshMsgFromResource(const std::string& resource, pcl_msgs::PolygonMesh& polygon_mesh_msg)
 {
-  shapes::Mesh* mesh = shapes::createMeshFromResource(resource);
-  if (!mesh)
-  {
-    ROS_ERROR_STREAM("Failed to load mesh from resource: '" << resource << "'");
-    return false;
-  }
-
-  shapes::ShapeMsg shape_msg;
-  if (!shapes::constructMsgFromShape(mesh, shape_msg))
-  {
-    ROS_ERROR_STREAM(__func__ << ": Failed to create shape message from mesh");
+  pcl::PolygonMesh pcl_mesh;
+  
+  if (!pcl::io::loadPolygonFile(resource, pcl_mesh)) {
```

Comment:
I think this function returns an integer (number of vertices), not a boolean, so your check should look like:
```suggestion
  if (pcl::io::loadPolygonFile(resource, pcl_mesh) < 0) {
```

Reply:
Alright I'll update this.

Reply:
Also, shouldn't it be a less than or equal to zero relationship because a mesh with zero vertices is not a viable mesh?

Reply:
This function should never return a value less than 0, and a failure will return a value of 0.

This function returns 0 when it errors and cannot identify the file format.  (See source file [here](https://github.com/PointCloudLibrary/pcl/blob/master/io/src/vtk_lib_io.cpp#L58).)  The rest of the time, it returns the output of one of the format-specific load functions.  Each of these returns the output of pcl::io::vtk2Mesh(), which returns 0 when it errors (see [here](https://github.com/PointCloudLibrary/pcl/blob/master/io/src/vtk_lib_io.cpp#L247) and [here](https://github.com/PointCloudLibrary/pcl/blob/master/io/src/vtk_lib_io.cpp#L254)).  Otherwise, in the event of a successful load, a positive integer value will be returned.

Using the unary operator `!` on any nonzero integer will result in 0 / false, and using the unary operator `!` on zero will result in 1/true.  Either the negation (`!`) or comparing to zero would be the correct way to check for failure here.